spec/enumerables_spec.rb

```diff
@@ -0,0 +1,273 @@
+      it 'should return true if all of the collection items evaluate to true' do
+        expect(array5.my_all? { |ele| ele >= 1 }).to be true
+        expect(range.my_all? { |ele| ele.is_a?(Numeric) }).to be true
+      end
+
+      it 'should return false if any of the collection items evaluates to false' do
+        expect(array3.my_all? { |ele| ele.is_a?(Integer) }).to be false
+        expect(range.my_all? { |ele| ele < 3 }).to be false
+      end
+    end
+
+    context 'when pattern is given as argument' do
+      it 'should return true if pattern === each element in collection' do
+        expect(array1.my_all?(Integer)).to be true
+        expect(array2.my_all?(/t/)).to be true
+      end
+
+      it 'should return false if pattern doesn\'t match each element in collection' do
+        expect(array3.my_all?(Integer)).to be false
+        expect(array1.my_all?(/e/)).to be false
+      end
+    end
+  end
+
+  describe '#my_any?' do
+    context 'when no block is given' do
+      context 'if  atleast one of elements is not (false) or (nil)' do
+        it 'should return true' do
+          expect(array3.my_any?).to be true
+        end
+      end
+
+      context 'if none of the elements is not (false) or (nil)' do
+        it 'should return false' do
+          expect(array4.my_any?).to be false
+        end
+      end
+    end
+
+    context 'when block given' do
+      it 'should return true if any of the collection items evaluates to true' do
+        expect(array5.my_any? { |ele| ele == 3 }).to be true
+        expect(range.my_any? { |ele| ele > 4 }).to be true
+      end
+
+      it 'should return false if none of the collection items evaluates to true' do
+        expect(array2.my_any? { |ele| ele.size > 6 }).to be false
+        expect(range.my_any? { |ele| ele > 5 }).to be false
+      end
+    end
+
+    context 'when pattern is given as argument' do
+      it 'should return true if pattern === any element in collection' do
+        expect(array3.my_any?(nil)).to be true
+        expect(array2.my_any?(/o/)).to be true
+      end
+
+      it 'should return false if pattern doesn\'t match any element in collection' do
+        expect(array2.my_any?(Integer)).to be false
+        expect(array2.my_any?(/s/)).to be false
+      end
+    end
+  end
+
+  describe '#my_none?' do
+    context 'when no block is given' do
+      context 'if none of the elements is not (false) or (nil)' do
+        it 'should return true' do
+          expect(array4.my_none?).to be true
+        end
+      end
+
+      context 'if  atleast one of elements is not (false) or (nil)' do
+        it 'should return false' do
+          expect(array3.my_none?).to be false
+        end
+      end
+    end
+
+    context 'when block given' do
+      it 'should return true if none of the collection items evaluates to true' do
+        expect(array5.my_none? { |ele| ele == 4 }).to be true
+        expect(range.my_none? { |ele| ele > 5 }).to be true
```

Comment:
* [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))